### ISSUE TITLE
medaka recipe is broken for newest version (1.2.1)

### ISSUE THREAD

**Issue by RaverJay** [bioconda/bioconda-recipes#26307]
I want to use medaka v1.2.1
samtools v1.10 is bugged, so I would want to get 1.9 or 1.11

`conda create -n medaka_test medaka=1.2.1`
grabs the bugged samtools v1.10, but builds the environment

`conda create -n medaka_test medaka=1.2.1 samtools=1.11`
conflicts:
```
Package samtools conflicts for:
samtools=1.11
medaka=1.2.1 -> samtools[version='>=1.9']
```

`conda create -n medaka_test medaka=1.2.1 samtools=1.9`
conflicts:
```
Package samtools conflicts for:
medaka=1.2.1 -> samtools[version='>=1.9']
samtools=1.9
```

This does not obviously make sense. There is a conflict, but conda create is incapable of actually showing it, it seems.

With mamba you can find what is wrong:
`mamba create -n medaka_test medaka=1.2.1 samtools=1.11`
```
Problem: package samtools-1.11-h6270b1f_0 requires htslib >=1.11,<1.12.0a0, but none of the providers can be installed
```

medaka 1.2.1 and samtools 1.11 do not mix:

medaka deps:
`conda search medaka=1.2.1 --info`
```
medaka 1.2.1 py38hfcf0ad1_0
---------------------------
file name   : medaka-1.2.1-py38hfcf0ad1_0.tar.bz2
name        : medaka
version     : 1.2.1
build       : py38hfcf0ad1_0
build number: 0
size        : 37.9 MB
license     : Mozilla Public License 2.0
subdir      : linux-64
url         : https://conda.anaconda.org/bioconda/linux-64/medaka-1.2.1-py38hfcf0ad1_0.tar.bz2
md5         : 483317cb802995cf3e1193e4707a85ee
timestamp   : 2020-11-26 23:36:19 UTC
dependencies: 
  - bcftools >=1.9
  - biopython
  - bzip2 >=1.0.8,<2.0a0
  - cffi
  - h5py
  - htslib >=1.10.2,<1.11.0a0
  - intervaltree >=3.0.0
  - libgcc-ng >=7.5.0
  - mappy
  - minimap2 >=2.17
  - numpy
  - ont-fast5-api
  - parasail-python
  - pysam >=0.16.0.1
  - pyspoa
  - python >=3.8,<3.9.0a0
  - python-edlib
  - python_abi 3.8.* *_cp38
  - requests
  - samtools >=1.9
  - tensorflow >=2.2.0
  - whatshap >=0.18
  - xz >=5.2.5,<5.3.0a0
  - zlib >=1.2.11,<1.3.0a0
```

samtools deps:
`conda search samtools=1.11 --info`
```
samtools 1.11 h6270b1f_0
------------------------
file name   : samtools-1.11-h6270b1f_0.tar.bz2
name        : samtools
version     : 1.11
build       : h6270b1f_0
build number: 0
size        : 383 KB
license     : MIT
subdir      : linux-64
url         : https://conda.anaconda.org/bioconda/linux-64/samtools-1.11-h6270b1f_0.tar.bz2
md5         : 185503bd7a14eadbc088a301784b22d1
timestamp   : 2020-10-04 13:29:15 UTC
dependencies: 
  - htslib >=1.11,<1.12.0a0
  - libgcc-ng >=7.5.0
  - ncurses >=6.2,<6.3.0a0
  - zlib >=1.2.11,<1.3.0a0
```

htslib conflicting.

But, the medaka recipe does **not** pose this requirement of   - htslib >=1.10.2,<1.11.0a0
https://github.com/bioconda/bioconda-recipes/blob/master/recipes/medaka/meta.yaml
```
...
    - htslib >=1.9
...
```

Could not find where this dep is coming from, is there a good way to search the tree?

**Comment by jmarshall** [comment]
That dep comes from Bioconda's htslib pinning: [bioconda_utils-conda_build_config.yaml](https://github.com/bioconda/bioconda-utils/blob/master/bioconda_utils/bioconda_utils-conda_build_config.yaml) pins htslib to 1.10.x, and 1.10.2 is what your medaka package has been built against.

Ways to fix this would be:

* Have the medaka recipe add a `conda_build_config.yaml` overriding the pinning so that it uses htslib 1.11 like samtools and bcftools do.

* As you figured out in https://github.com/nanoporetech/medaka/issues/186#issuecomment-758567703, there is a small patch to htslib 1.10 (already present in 1.11) that would fix the samtools problem you're encountering. So bioconda could add a htslib/1.10.2 recipe that makes a new htslib-1.10.2 package that incorporates this patch. This would make the existing medaka package work.

* Bioconda could update the htslib pinning and rebuild the medaka package.

HTSlib et al 1.11 have been out for several months and it is probably about time bioconda updated its pinning. However 1.12 appears to be perhaps weeks away, so it would probably be worthwhile for @bioconda/core to wait for that before updating the pinning and rebuilding.

It's probably not worthwhile to do the work to retropatch a fixed htslib-1.10.2 bioconda build and package, as these fixes already exist in htslib-1.11 and later. So while they wait for the pinning to be updated, medaka should probably add a `conda_build_config.yaml` and bump their package to fix this in the meantime.

**Comment by RaverJay** [comment]
Thanks for looking into this!

**Comment by jmarshall** [comment]
This was fixed by PR #26340.
